sql/sql_statistics.cc

```diff
@@ -1580,6 +1593,60 @@ class Histogram_builder
   }
 };
 
+class Histogram_builder_json : public Histogram_builder
+{
+std::vector<std::string> bucket_bounds;
+
+public:
+  Histogram_builder_json(Field *col, uint col_len, ha_rows rows)
+  : Histogram_builder(col, col_len, rows)
+  {
+    Column_statistics *col_stats= col->collected_stats;
+    min_value= col_stats->min_value;
+    max_value= col_stats->max_value;
+    histogram= &col_stats->histogram;
+    hist_width= histogram->get_width();
+    bucket_capacity= (double) records / (hist_width + 1);
+    curr_bucket= 0;
+    count= 0;
+    count_distinct= 0;
+    count_distinct_single_occurence= 0;
+    bucket_bounds = {};
+  }
+
+  ~Histogram_builder_json() override = default;
+
+  int next(void *elem, element_count elem_cnt) override
+  {
+    count_distinct++;
+    if (elem_cnt == 1)
+      count_distinct_single_occurence++;
+    count+= elem_cnt;
+    if (curr_bucket == hist_width)
+      return 0;
+    if (count > bucket_capacity * (curr_bucket + 1))
+    {
+      column->store_field_value((uchar *) elem, col_length);
+      StringBuffer<MAX_FIELD_WIDTH> val;
+      column->val_str(&val);
+      bucket_bounds.emplace_back(val.c_ptr());
+      curr_bucket++;
+    }
+    return 0;
+  }
+
+  void build() {
+    Json_writer *writer = new Json_writer();
```

Comment:
This seems to result in a memory leak (`mtr` reports a "Memory not freed" warning) and I'm not sure of how to fix that yet.

So far, I've tried:
- freeing up the memory allocated to `JSON_writer` with
```cpp
delete writer;
writer = NULL;
```

- replacing the `JSON_writer` pointer with a smart pointer i.e
```cpp
std::unique_ptr<Json_writer> writer(new Json_writer());
```

They both remove the memory leak warning, but also result in garbage histogram values stored in the database.